rust/worker/src/segment/metadata_segment.rs

```diff
@@ -420,7 +484,289 @@ impl<'a> SegmentWriter<'a> for MetadataSegmentWriter {
                             None => {}
                         },
                         None => {}
+                    };
+                }
+                Operation::Delete => match &record.0.data_record {
+                    Some(data_record) => {
```

Comment:
not a comment for this PR necessarily - I don't like the deep nesting. Makes it hard to read. I wonder if there are any Rust specific guidelines on this